src/main/java/org/traccar/protocol/Xexun2ProtocolDecoder.java

```diff
@@ -176,7 +176,19 @@ protected Object decode(
                         position.set(Position.KEY_SATELLITES, buf.readUnsignedByte());
                         position.setLongitude(convertCoordinate(buf.readDouble()));
                         position.setLatitude(convertCoordinate(buf.readDouble()));
-
+                    }
+                    if (BitUtil.check(positionMask, 7)) {
+                        buf.skipBytes(2); // length = 30
+                        buf.skipBytes(1); // marker = 'G'
+                        buf.skipBytes(2); // length = 27
```

Comment:
I think these needs to be handled properly. You need to honor the length fields. And also probably we need to check for the marker. They clearly intend this to be extendable, even if for now it's fixed.